apps/api/v2/src/ee/event-types/event-types.repository.ts

```diff
@@ -0,0 +1,44 @@
+import { CreateEventTypeInput } from "@/ee/event-types/inputs/create-event-type.input";
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { Injectable, NotFoundException } from "@nestjs/common";
+
+import { getEventTypeById } from "@calcom/platform-libraries";
+
+@Injectable()
+export class EventTypesRepository {
+  constructor(private readonly dbRead: PrismaReadService, private readonly dbWrite: PrismaReadService) {}
+
+  async createUserEventType(userId: number, body: CreateEventTypeInput) {
+    return this.dbWrite.prisma.eventType.create({
+      data: {
+        ...body,
+        userId,
```

Comment:
is userId not set by the following connect call?

Reply:
`EventType.userId` is used to reference `EventType.owner` (one to one) and connect is for `EventType.users` (one to many)